From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evgeny Karpov <eukarpov@gmail.com>
Date: Fri, 15 Nov 2024 13:51:42 +0100
Subject: [PATCH 02/29] aarch64: Add stdcall and cdecl attributes

This patch adds stdcall and cdecl attributes, which might be used for
DLL export/import in MinGW.

gcc/ChangeLog:

	* config/aarch64/aarch64.cc: Update.

(cherry picked from commit 7f4c63742c8c20e1a33244d30a75406bd6817723)
---
 gcc/config/aarch64/aarch64.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gcc/config/aarch64/aarch64.cc b/gcc/config/aarch64/aarch64.cc
index 5544d5f8c7a..abd207fb3dd 100644
--- a/gcc/config/aarch64/aarch64.cc
+++ b/gcc/config/aarch64/aarch64.cc
@@ -935,6 +935,8 @@ static const attribute_spec aarch64_gnu_attributes[] =
   { "Advanced SIMD type", 1, 1, false, true,  false, true,  NULL, NULL },
   { "SVE type",		  3, 3, false, true,  false, true,  NULL, NULL },
   { "SVE sizeless type",  0, 0, false, true,  false, true,  NULL, NULL },
+  { "cdecl",		  0, 0, false, false, false, false, NULL, NULL },
+  { "stdcall",		  0, 0, false, false, false, false, NULL, NULL },
 #if TARGET_DLLIMPORT_DECL_ATTRIBUTES
   { "dllimport", 0, 0, false, false, false, false, handle_dll_attribute, NULL },
   { "dllexport", 0, 0, false, false, false, false, handle_dll_attribute, NULL },
